### PR TITLE
better log2 doc

### PR DESCRIPTION
**Pull request by pkgoogle**
better log2 doc

Part of https://github.com/google/jax/issues/21461

### CODE REVIEW

File: jax/_src/numpy/ufuncs.py [hunk 1]
```diff
@@ -827,9 +827,33 @@ def _logaddexp2_jvp(primals, tangents):
...
+    >>> jnp.log2(x1)
+    Array([-2., -1.,  0.,  1.,  2.,  3.], dtype=float32)
+
+    >>> x2 = jnp.array([-0.25, -0.5, -1, -2, -4, -8])
+    >>> jnp.log2(x2)
+    Array([nan, nan, nan, nan, nan, nan], dtype=float32)
+
+    >>> x3 = jnp.array([1+1j, -1+0j, 1+0j, 2+2j])
+    >>> jnp.log2(x3)
+    Array([0.49999994+1.13309j, 0.        +4.53236j, 0.        +0.j     ,
```
Comment (jakevdp) [inline jax/_src/numpy/ufuncs.py:854]: This complex one is going to be problematic due to float accuracy; I'd probably just remove it.

**Comment by jakevdp** [comment]
Thanks! If you can squash the changes into a single commit, then we'll get this merged!

File: jax/_src/numpy/ufuncs.py [hunk 2]
```diff
@@ -827,9 +827,28 @@ def _logaddexp2_jvp(primals, tangents):
...
+    to inexact dtype.
+
+  Examples:
+    >>> x1 = jnp.array([0.25, 0.5, 1, 2, 4, 8])
+    >>> jnp.log2(x1)
+    Array([-2., -1.,  0.,  1.,  2.,  3.], dtype=float32)
+
+    >>> x2 = jnp.array([-0.25, -0.5, -1, -2, -4, -8])
+    >>> jnp.log2(x2)
+    Array([nan, nan, nan, nan, nan, nan], dtype=float32)
```
Comment (dfm) [inline jax/_src/numpy/ufuncs.py:850]: I'm not sure that this example really adds much to the discussion. I'd recommend that we remove it!

**Review by dfm — CHANGES_REQUESTED** [review]
I have one last quick request here, then let's get this merged. Thanks!

**Review by dfm — APPROVED** [review]
Looks good - thanks!
